### PR TITLE
Add support for 'docker version' and 'podman version'

### PR DESCRIPTION
**Pull request by LewisGaul**
Fixes https://github.com/gabrieldemarmiesse/python-on-whales/issues/591

- [x] Resolve what to do about the added requirement for pydantic v2
   - Making it work with pydantic v1 for now (although podman's `APIVersion` and `BuiltTime` fields won't be recognised with pydantic v1)
   - Raised separate issue for removing pydantic v1 support: https://github.com/gabrieldemarmiesse/python-on-whales/issues/594
- [x] Add tests

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
I would be happy to drop support for pydantic V1 when it's end of life, so in one month. If this PR is not a pressing matter, we can do it at this point in time :)

File: pyproject.toml [hunk 1]
```diff
@@ -37,3 +37,4 @@ markers = [
     "podman: marks tests as needing a podman engine to run"
 ]
 xfail_strict = true
+pythonpath = ["./"]
```
Comment (LewisGaul) [inline pyproject.toml:40]: This makes just `pytest` work from the project root, rather than needing `python3 -m pytest`

**Review by gabrieldemarmiesse — COMMENTED** [review]
Looks good as usual! We can merge when pydantic v1 is EOL

**Review by gabrieldemarmiesse — APPROVED** [review]

**Comment by LewisGaul** [comment]
> Looks good as usual! We can merge when pydantic v1 is EOL

To be clear, I updated it to be compatible with pydantic v1 so that we don't need to wait.

**Comment by gabrieldemarmiesse** [comment]
Ho my bad, I went over it a bit too quickly since I already read the diff once. Let's merge it now :)

**Pull request merged**